### PR TITLE
TCM-734 | Add UserId for Amplitude 

### PR DESCRIPTION
**Pull request by vespertineSiren**
### Summary

- Inject analytics manager to backendmanager, call `AnalyticsManager.setUserId(id: String)` in `BackendManager.user()`.

### Issue

[TCM-734](https://hyfnla.atlassian.net/browse/TCM-734)

### CODE REVIEW

**Comment by dkhamsing** [comment]
@vespertineSiren in ios we set the amplitude user id in BackendManager.user (similar idea to xpass where it is done in BackendManager.profile).. is that something we can do in android

File: app/src/main/java/com/trx/consumer/screens/home/HomeViewModel.kt [hunk 1]
```diff
@@ -60,6 +60,7 @@ class HomeViewModel @ViewModelInject constructor(
             backendManager.user()
             eventShowHud.postValue(false)
             cacheManager.user()?.let { user ->
+                analyticsManager.setUserId(user.uid)
```
Comment (onodude) [inline app/src/main/java/com/trx/consumer/screens/home/HomeViewModel.kt:63]: shouldn't this be in the backend manager where we do: `IAPManager.shared.identify(user.uid)`

**Review by onodude — CHANGES_REQUESTED** [review]
there are lots of changes?

File: app/src/main/java/com/trx/consumer/screens/home/HomeViewModel.kt (hunk 1 above)
Comment (vespertineSiren) [reply app/src/main/java/com/trx/consumer/screens/home/HomeViewModel.kt:63]: Yeah just clarified it with @dkhamsing over slack. Will revert.

**Comment by vespertineSiren** [comment]
> there are lots of changes?

A lot of them are ktlint related. Will revert.

File: app/src/main/java/com/trx/consumer/screens/home/HomeViewModel.kt (hunk 1 above)
Comment (vespertineSiren) [reply app/src/main/java/com/trx/consumer/screens/home/HomeViewModel.kt]: Here you go: https://github.com/hyfn/trx-consumer-android/pull/170/commits/e40226c4256ff732a898b71d80af4e92a2213998

**Review by onodude — APPROVED** [review]
thanks!

**Pull request merged**